keps/sig-node/2053-downward-api-hugepages/README.md

```diff
@@ -0,0 +1,230 @@
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+    - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+This KEP exposes hugepages in the downward API.
+
+## Motivation
+
+Pods are unable to know their hugepage request or limits via the downward API.  HugePages
+are a natively supported resource in Kubernetes and should be visible in downward API
+consistent with other resources like cpu, memory, ephemeral-storage.
+
+### Goals
+
+- Add support for hugepage requests and limits for all page sizes in downward API
+
+### Non-Goals
+
+- Change any other aspect of hugepage support
+
+## Proposal
+
+Define a new feature gate: `DownwardAPIHugePages`.
+
+If enabled, the `kubelet` will expose hugepage request and limits into the
+pod via the downward API.
+
+### Risks and Mitigations
+
+The primary risk for this proposal is that it loosens validation for
+Pods accepted by the API server.  As a result, the `kube-apiserver` should
+not enable support for this feature until all known `kubelet` in the cluster
```

Comment:
Since this is not a new field, but is a newly accepted value for an **existing** field, clarify what the gate protects. I would expect:

* With the gate enabled, the newly allowed values are permitted in all creation and update scenarios.
* When the gate disabled, the new values are permitted only in updates of objects which already contain the new values. Use in creation or in updates of objects which do not already use the new values will fail validation.